src/inference/include/openvino/runtime/intel_gpu/properties.hpp

```diff
@@ -0,0 +1,128 @@
+static constexpr Property<std::string, PropertyMutability::RO> uarch_version{"GPU_UARCH_VERSION"};
+
+/**
+ * @brief Read-only property to get count of execution units for current GPU
+ */
+static constexpr Property<uint32_t, PropertyMutability::RO> execution_units_count{"GPU_EXECUTION_UNITS_COUNT"};
+
+/**
+ * @brief Read-only property to get statistics of GPU memory allocated by engine for each allocation type
+ * It contains information about current memory usage
+ */
+static constexpr Property<std::map<std::string, uint64_t>, PropertyMutability::RO> memory_statistics{
+    "GPU_MEMORY_STATISTICS"};
+
+/**
+ * @brief Read-only property to get maximum batch size which does not cause performance degradation due to memory swap
+ * impact.
+ */
+static constexpr Property<uint32_t, PropertyMutability::RO> max_batch_size{"GPU_MAX_BATCH_SIZE"};
+
+/**
+ * @brief Turning on this key enables to unroll recurrent layers such as TensorIterator or Loop with fixed iteration
+ * count. This key is turned on by default. Turning this key on will achieve better inference performance for loops with
+ * not too many iteration counts (less than 16, as a rule of thumb). Turning this key off will achieve better
+ * performance for both graph loading time and inference time with many iteration counts (greater than 16). Note that
+ * turning this key on will increase the graph loading time in proportion to the iteration counts.
+ * Thus, this key should be turned off if graph loading time is considered to be most important target to optimize.*/
+static constexpr Property<bool> enable_loop_unrolling{"GPU_ENABLE_LOOP_UNROLLING"};
+
+namespace hint {
+/**
+ * @brief This key instructs the GPU plugin to use throttle hints the OpenCL queue throttle hint
+ * as defined in https://www.khronos.org/registry/OpenCL/specs/opencl-2.1-extensions.pdf,
+ * chapter 9.19. This option should be used with an unsigned integer value (1 is lowest energy consumption)
+ * 0 means no throttle hint is set and default queue created.
+ */
+static constexpr Property<uint32_t> queue_throttle{"GPU_QUEUE_THROTTLE"};
+
+/**
+ * @brief This key instructs the GPU plugin to use the OpenCL queue priority hint
+ * as defined in https://www.khronos.org/registry/OpenCL/specs/opencl-2.1-extensions.pdf
+ * this option should be used with an unsigned integer value (1 is lowest priority)
+ * 0 means no priority hint is set and default queue is created.
+ */
+static constexpr Property<uint32_t> queue_priority{"GPU_QUEUE_PRIORITY"};
+
+/**
+ * @brief Enum to define possible host task priorities which is set cpu core type of TBB affinity used in load network.
+ * It is only affected on Hybrid CPUs. If the device doesn't support Hybrid CPUs, it is set to the ANY.
+ */
+enum class HostTaskPriority {
+    LOW = 0,   //!< Mapped to IStreamsExecutor::Config::LITTLE (E-cores)
+    HIGH = 1,  //!< Mapped to IStreamsExecutor::Config::BIG (P-cores)
+    ANY = 2,   //!< Mapped to IStreamsExecutor::Config::ANY (Any cores)
```

Comment:
`IStreamsExecutor::Config::ANY` is an internal API, can we have generic description w/o reference to internal API?

Reply:
Replaced with `ov::hint::Priority` and updated the description of ov::intel_gpu::hint::host_task_priority property